core/src/main/java/org/mule/api/context/notification/AbstractBlockingServerEvent.java

```diff
@@ -7,7 +7,15 @@
 
 package org.mule.api.context.notification;
 
-public class AbstractBlockingServerEvent extends ServerNotification implements BlockingServerEvent
+import org.mule.api.MuleMessage;
+
+/**
+ * Default implementation of {@link @BlockingServerEvent} that guarantees all the notifications that extends this class
+ * sends a copy of the {@link MuleMessage} instead of the original one for avoiding race conditions for getting the message ownership.
```

Comment:
...the original one. This avoids race conditions when getting the message ownership.